BCnEnc.Net/Shared/EncodedBlocks.cs

```diff
@@ -64,11 +66,10 @@ internal unsafe struct Bc1Block
 		}
 	}
 
-
 	[StructLayout(LayoutKind.Sequential)]
 	internal unsafe struct Bc2Block
 	{
-		public ulong alphaColors;
+		public Bc2AlphaBlock alphaBlock;
```

Comment:
I like what you've done here. Good code re-use. Maybe the block part of bc4block could be named something more generic(BcSingleComponentBlock?) and bc4block could be changed to be just a wrapper around it? The encoding for that block can be extracted to its own class as well. Used in bc4, bc3 alpha, bc5 and atci alpha.

Reply:
I'm actually not sure what you want to express here. What should I wrap into what? I didn't fully understand.

Reply:
Umm, since the current bc4block is used as an alpha block in two formats, and bc5 is just two bc4 blocks, the actual bc4block could be separated into its own BcSingleComponentBlock. Then change the others to use that and change Bc4Block to just be a wrapper around the SingleComponentBlock. The RedAsLuminance stuff is bc4 specific though, so that should be in the bc4block instead of the generic block.

Reply:
Ah, ok. Now I understand. However the encoding in this block is BC4 specific as well. It isn't defined in any other way, so even if we extract the logic out, it would need to be named after BC4, where it is specified.

Reply:
Ah, maybe it can stay as it is then. I might take a look at it again later.

Reply:
I do understand the thought, and I think I have an idea. I will commit it and if you don't like it, we just revert it again.